### ISSUE TITLE
Event stream object

### ISSUE THREAD

**Issue by ottiker** [jillix/flow-nodejs#152]
To solve the problem of callback functionality for events, we could use the technique of the websocket `link` object to have a "stream" capability.
Example:

``` js
// create the event
var event = this.emit('eventName');
// ..or
var event = this.emit({
    event: 'name',
    all: false,
    to: 'instanceName'
});

// add a data listener
event.data(function (err, data) {});
// send data to stream
event.send(err, {/* data */});
// destroy the event
event.end(err, {/* data */});
```

That would mean, that every listener method gets the `event` stream object and could broadcast data to all other listeners through the `event` object. Since an `event` object is created on every `emit` the "callback" context is still valid.

Things that would needed to be updated:
- composition `flow.out` config
- `engine.flow` listener code
- all current `emit` calls

**Comment by ottiker** [comment]
The `link` object could also be used for `http` requests:
- `link.data` -> receive the request body
- `link.send` -> write to the response stream `res.write();`
- `link.end` -> end the response `res.end();`

That would make the `event`, `ws` and `http` handlers consistent.

**Comment by ottiker** [comment]
Was strangely closed, when the `v0.1.0-beta` branch was merged to master.

**Comment by ottiker** [comment]
Info: Issue #33 was also about callback events.
